### PR TITLE
Add allowances in railjson generator

### PR DESCRIPTION
**Pull request by Castavo**

### CODE REVIEW

File: python/railjson_generator/railjson_generator/schema/simulation/train_schedule.py [hunk 1]
```diff
@@ -26,6 +37,31 @@ def add_stop(self, *args, **kwargs):
...
+        self.allowances.append(allowance)
+
+    def add_standard_single_value_allowance(self, value_kind: str, value: float, distribution: str = "LINEAR"):
+        """Add a standard allowance with a single value. For more information on allowances, see
+        the documentation of the Allowance class in osrd_schemas."""
+        if value_kind == "time":
+            value = AllowanceTimeValue(seconds=value)
+        elif value_kind == "time_per_distance":
+            value = AllowanceTimePerDistanceValue(minutes=value)
+        elif value_kind == "percent":
```
Comment (Erashin) [inline python/railjson_generator/railjson_generator/schema/simulation/train_schedule.py]: Might be better to stay coherent with the rest of the code which uses `"percentage"` (see AllowancePercentValue). Or use `percent` everywhere.

File: python/railjson_generator/railjson_generator/scripts/examples/small_infra.py [hunk 2]
```diff
@@ -627,13 +627,16 @@ def add_signal_on_ports(switch, ports: Mapping[str, Tuple[str, str]]):
     stops=[Stop(location=loc, duration=60) for loc in stop_locations],
 )
 
+train_0.add_standard_single_value_allowance("percent", 5)
```
Comment (Erashin) [inline python/railjson_generator/railjson_generator/scripts/examples/small_infra.py]: Small_infra will have everything except `AllowanceTimeValue`. Replace one of the 2 `AllowancePercentValue` (which test Linear and Mareco cases) to test the time value, or add it to another train (if there is another one in small_infra that is).

File: python/railjson_generator/railjson_generator/scripts/examples/small_infra.py
Comment (Erashin) [inline python/railjson_generator/railjson_generator/scripts/examples/small_infra.py:1]: There are a couple of tests missing in tests/test_train_schedule.py. Might be a good time to add them (only Mareco is tested atm for example).

**Review by Erashin — COMMENTED** [review]
LGTM :)

File: python/railjson_generator/railjson_generator/scripts/examples/small_infra.py
Comment (Castavo) [reply python/railjson_generator/railjson_generator/scripts/examples/small_infra.py:1]: I think it's out of the scope of this PR and I don't know if we want to link the integration tests to the generated schedules (maybe we do, but I don't want this PR to drag on).  
If you meant just writing tests for schedules written by hand, I'm pretty sure it's out of scope.

If you want to add those tests please fell free to do it

Comment (Castavo) [reply python/railjson_generator/railjson_generator/scripts/examples/small_infra.py:1]: I think it's out of the scope of this PR and I don't know if we want to link the integration tests to the generated schedules (maybe we do, but I don't want this PR to drag on). If you meant just writing tests for schedules written by hand, I'm sure it's out of scope.

If you want to add those tests please fell free to do it

**Review by Erashin — APPROVED** [review]

**Pull request merged**